.github/workflows/publish.yml

```diff
@@ -9,31 +9,18 @@ jobs:
     name: Build and push to PyPI
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Prep Python
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: "3.11"
 
-      - name: Install build tools
-        run: |
-          python -m pip install build setuptools twine wheel --user
-        continue-on-error: false
-      - name: Build source and wheel
-        id: build
-        run: |
-          python -m build --outdir dist/
-        continue-on-error: true
-      - name: Check the output
-        run: |
-          python -m twine check --strict dist/*
-        continue-on-error: false
-      - name: Die on failure
-        if: steps.build.outcome != 'success'
-        run: exit 1
+      - name: Build
+        run: pipx run build
```

Comment:
`pipx` is pre-installed on GHA so we can use it for running `build` in one go, which builds an `sdist` and then builds a wheel from the `sdist`. The `Install build tools` step was entirely useless because `build` [builds the projects with `pyproject.toml + [build-system]` in a new isolated environment.](https://pip.pypa.io/en/stable/reference/build-system/pyproject-toml/)